feeds/pypi/pypi.go

```diff
@@ -73,28 +87,97 @@ func fetchPackages() ([]*Package, error) {
 	return rssResponse.Packages, nil
 }
 
+func fetchCriticalPackages(packageList []string) ([]*Package, error) {
+	responseChannel := make(chan *Response)
+	errs := make(chan error)
+	var wg sync.WaitGroup
+
+	for _, pkgName := range packageList {
+		wg.Add(1)
+		go func(pkgName string) {
+			resp, err := httpClient.Get(fmt.Sprintf(packageURLFormat, pkgName))
+			if err != nil {
+				errs <- err
+				return
+			}
+			defer resp.Body.Close()
+			rssResponse := &Response{}
+			err = xml.NewDecoder(resp.Body).Decode(rssResponse)
+			if err != nil {
+				errs <- err
+				return
+			}
+			wg.Done()
+			responseChannel <- rssResponse
+		}(pkgName)
+	}
+	wg.Wait()
+	select {
+	case err := <-errs:
+		return nil, err
+	default:
+		close(errs)
+	}
+
+	pkgs := []*Package{}
+	for i := 0; i < len(packageList); i++ {
+		select {
+		case response := <-responseChannel:
+			pkgs = append(pkgs, response.Packages...)
+		default:
+			return nil, errMissingParallelPackageResponse
+		}
+	}
+	return pkgs, nil
+}
+
 type Feed struct {
+	packages *[]string
+
 	lossyFeedAlerter *feeds.LossyFeedAlerter
 }
 
-func New(eventHandler *events.Handler) *Feed {
+func New(feedOptions feeds.FeedOptions, eventHandler *events.Handler) (*Feed, error) {
 	return &Feed{
+		packages:         feedOptions.Packages,
 		lossyFeedAlerter: feeds.NewLossyFeedAlerter(eventHandler),
-	}
+	}, nil
 }
 
 func (feed Feed) Latest(cutoff time.Time) ([]*feeds.Package, error) {
 	pkgs := []*feeds.Package{}
-	pypiPackages, err := fetchPackages()
+	var pypiPackages []*Package
+	var err error
+
+	if feed.packages == nil {
+		// Firehose fetch all packages.
+		pypiPackages, err = fetchPackages()
+	} else {
+		// Fetch specific packages individually from configured packages list.
+		pypiPackages, err = fetchCriticalPackages(*feed.packages)
+	}
+
 	if err != nil {
-		return pkgs, err
+		return nil, err
 	}
 	for _, pkg := range pypiPackages {
-		pkg := feeds.NewPackage(pkg.CreatedDate.Time, pkg.Name(), pkg.Version(), FeedName)
+		pkgName, err := pkg.Name()
+		if err != nil {
+			return nil, err
+		}
+		pkgVersion, err := pkg.Version()
+		if err != nil {
+			return nil, err
+		}
+		pkg := feeds.NewPackage(pkg.CreatedDate.Time, pkgName, pkgVersion, FeedName)
 		pkgs = append(pkgs, pkg)
 	}
 	feed.lossyFeedAlerter.ProcessPackages(FeedName, pkgs)
```

Comment:
Checking for overlaps could be a little redundant when configured for `CriticalPackages` as every configured package would have had to have 50 (I assume the package release RSS is also limited to 50) releases in-between a polling event to not cause an overlap. It may make more sense to check old-new results on a per package basis with this in mind, or not at all.